apps/docs/stories/architecture-decision-records/0013-why-do-we-need-to-convert-carousel-item.stories.mdx

```diff
@@ -0,0 +1,80 @@
+
+# ADR 0013: Why do we need to convert carousel item
+
+🗓️ 2023-08 · ✍️ [@schirrel](https://github.com/schirrel), [@MayaraRMA](https://github.com/MayaraRMA), [@tassioFront](https://github.com/tassioFront) and [@felipefialho](https://twitter.com/felipefialho_)
+
+## Context
+
+Wrapping up Swiper HTMLElements as Atomium element.
+
+## Problems
+
+When creating `<atom-carousel>` and its items `<atom-carousel-item>`, the HTML three place the `<atom-carousel-item>` inside `<swiper-element>`, but the swiper needs its children to be `<swiper-slide>`.
+
+Then we need to convert each `atom-carousel-item` into `swiper-slide`.
+
+## Considered Options
+
+To convert the inner `<atom-carousel-item>` to `<swiper-slide>`, and we have two options:
+
+1. Create and `atom-carousel-item` component and inside itself creating the `<swiper-slide>` component on parent and them self deleting from parent.
+2. The parent itself convert all children at once.
+
+## Decision Drivers
+
+- Performance
+- Complexity
+- Compatibility
+
+### Performance
+
+We had performed a test with both approach using 9 items inside the Carousel:
+
+- With children `atom-carousel-item` doing the conversion, it took _0.50ms_
+- With parent `atom-carousel` doing the conversion, it took _0.20ms_
+
+### Complexity
+
+Both codes shows same complexibility:
+First approach - parent conversion :
+
+```js
+  private convertChildren() {
+    this.swiperEl.innerHTML = this.host.innerHTML.replace(
+      /atom-carousel-item/g,
+      'swiper-slide'
+    )
+    let idx = 0
+    while (idx < this.host.children.length) {
+      if (this.host.children[idx].outerHTML.indexOf('atom-carousel-item') > -1)
+        this.host.removeChild(this.host.children[idx])
+      else idx++
+    }
+  }
+```
+
+Second approach - child conversion:
+
+```js
+const parentSwiperContainer =
+  this.host.parentElement.querySelector('swiper-container')
+const swiperSlide = this.host.querySelector('swiper-slide')
+parentSwiperContainer.appendChild(swiperSlide)
+this.host.parentElement.removeChild(this.host)
+```
+
+Taking in consideration the complexity of both codes, not only the parent's conversion approach was using less code but take also avoid us to create a component only to destruct itself. This also avoid the have one more file to maintain, build, test and also to be on final bundle.
+
+### Compatibility
+
+This subject was in fact the decision maker, even tho the performance of the first approach took half of the time, when it came to compatibility we discovered why.
+For simples cases, such static card, plain text or image showcases, no difference between both, but when we started to use with complex components, that had to deal with framework functions such reactivity we face a bigger issue. The HTMLString conversion component loose _"connections "_ with the inner content. So it just became a plain HTML String copy, instead of a state and object reference.
```

Comment:
```suggestion

For simples cases, such static card, plain text or image showcases, no difference between both, but when we started to use with complex components, that had to deal with framework functions such reactivity we face a bigger issue. The HTMLString conversion component loose _"connections "_ with the inner content. So it just became a plain HTML String copy, instead of a state and object reference.
```